sdk/core/azure-core-v2/src/main/java/com/azure/v2/core/implementation/AccessTokenCache.java

```diff
@@ -71,69 +71,64 @@ public AccessToken getToken(TokenRequestContext tokenRequestContext, boolean che
     }
 
     private AccessToken retrieveToken(TokenRequestContext tokenRequestContext, boolean forceFetchToken) {
+        validateTokenRequestContext(tokenRequestContext);
+        return fetchToken(tokenRequestContext, forceFetchToken);
```

Comment:
I would just put `validateTokenRequestContext` inside `fetchToken` (Or vice versa depending on if you like the name `fetchToken` or `retrieveToken` more.)